sentences/de/homeassistant_HassGetCurrentTime.yaml

```diff
@@ -5,4 +5,5 @@ intents:
       - sentences:
           - "wie spät ist es"
           - "(wie viel|wieviel) uhr (ist es|haben wir)[ jetzt]"
-          - "(nenn[e]|gib|gebe) mir die uhrzeit"
+          - "(nenn[e]|gib|gebe|sag[e]) mir die [uhr]zeit"
+          - "was (ist|haben wir) für zeit"
```

Comment:
this sounds incorrect "was ist für zeit" or "was haben wir für zeit" are even incomplete sentences, nor are they any common shorthand sentences

Reply:
I don't know if it is correct from a German point of view, but this is how Swiss people would say it (in high German, not Swiss dialect).

Reply:
at least "was haben wir für EINE zeit" might be ok, but still "was ist für eine zeit" sounds wrong
@andreasbrett what do you think?

Reply:
While I certainly can't speak from a swiss POV, I am unsure if "Schweizer Hochdeutsch" should be included in `de` or rather in `de-CH`. I think adding "Schweizer Hochdeutsch" to `de` would induce a lot of new sentences that in the long run would IMHO require being outsourced to a discrete language (I know, I'm very finicky with this).

From what I could gather Schweizer Hochdeutsch is mostly used in writing. Also my experience in communication with Swiss people so far has been that they would talk Schweizerdeutsch unless they had to switch to Schweizer Hochdeutsch so I could understand them. But again this is very very anecdotic. So I'm not sure if adding Schweizer Hochdeutsch brings that much value.

From those both options though to me "was haben wir für EINE zeit" sounds more reasonable and could potentially also be used by german speakers. "was ist für eine zeit" sounds very dialectic to me - but again, not much experience with Schweizer Hochdeutsch.

Reply:
https://de.wikipedia.org/wiki/Schweizer_Hochdeutsch#M%C3%BCndliche_Verwendung

> In Alltagssituationen wird Schweizer Hochdeutsch grösstenteils nur mit Menschen gesprochen, die den Dialekt nicht verstehen. Unter den verschiedenen Dialekten herrscht weitgehende gegenseitige Verständlichkeit, so dass hier nicht auf das Schweizer Hochdeutsch zurückgegriffen werden muss. In der Schweiz haben Dialekte zumeist ein höheres Ansehen als im übrigen deutschen Sprachraum. Die mündliche Umgangssprache zwischen Deutschschweizern ist fast ausnahmslos die jeweils heimische Mundart, der Ortsdialekt, unabhängig von Bildung und dem gesellschaftlichen Status.

Reply:
> While I certainly can't speak from a swiss POV, I am unsure if "Schweizer Hochdeutsch" should be included in `de` or rather in `de-CH`. I think adding "Schweizer Hochdeutsch" to `de` would induce a lot of new sentences that in the long run would IMHO require being outsourced to a discrete language (I know, I'm very finicky with this).
> 
> From what I could gather Schweizer Hochdeutsch is mostly used in writing. Also my experience in communication with Swiss people so far has been that they would talk Schweizerdeutsch unless they had to switch to Schweizer Hochdeutsch so I could understand them. But again this is very very anecdotic. So I'm not sure if adding Schweizer Hochdeutsch brings that much value.
> 
> From those both options though to me "was haben wir für EINE zeit" sounds more reasonable and could potentially also be used by german speakers. "was ist für eine zeit" sounds very dialectic to me - but again, not much experience with Schweizer Hochdeutsch.

Basically agree, that most Swiss would prefer speaking dialect before speaking high german. For me the issue popped up, as for "Schweizerdeutsch"  there is not (yet?) a STT available in HA. Everything is transcribed to "Hochdeutsch" (not sure, which variant, but I would assume "de_de"). So despite the availability of the sentences in Swiss dialect, "Schweizerdeutsch" cannot be used in HA over STT. I therefore tried out the German setting and apparently talked Swiss high german. This is how I noticed that some (for me common) sentences are not available.

I appreciate that the use case might not be huge. On the other hand, I would not expect a material delay due to the additions. And it would at least enable the usage of STT for (german speaking) Swiss users.

Regarding the distinction of "de" and "de_CH", please note that in HA "de_CH" is used for dialect (as published [here](https://home-assistant.github.io/intents/), unfortunately in the configuration dialog "de_CH" is translated as "Schweizer Hochdeutsch"). "Schweizer Hochdeutsch" has much more common with "Hochdeutsch" than with "Schweizerdeutsch". So if we could integrate "Schweizer Hochdeutsch" somewhere, it would make more sense in "de". I would regret if this should not be possible, but have to leave this up to you.

Regarding the sentences: I would never say "was ist für **eine** zeit", but "was ist für zeit". For me "Zeit" is not countable, you cannot have one or two times. Of course this might be different in Germany (perhaps also depending on the region). But if you would only add it with "eine", then perhaps better leave it at all.

Reply:
Oh what a mess. The HA Cloud TTS models for `Schweizer Hochdeutsch` (= `Deutsch (Schweiz)` in the underlying Azure speech services) are mixed as well.

- Jan is speaking Schweizer Hochdeutsch (I would even say it is German Hochdeutsch...)
- Leni is speaking Schweizerdeutsch

The STT on the other hand transcribes (as you mentioned) to German Hochdeutsch:
https://azure.microsoft.com/de-de/updates?id=release-of-swiss-german-dialect-speech-to-text

I don't know how this could be circumvented, as we'd want to have Schweizerdeutsch sentences to match against both text and audio pipelines but as of now both pipelines provide different languages. Back to the original question: I think this is a rather foundational problem that needs solving and is far beyond the initial question regarding "was ist für zeit". Maybe time to make this a thread in the language leader board.